From: Max Gurtovoy <maxg@mellanox.com>
Date: Mon, 6 Nov 2017 16:18:51 +0200
Subject: nvme-rdma: fix nvme_rdma_create_queue_ib error flow
Patch-mainline: v4.15-rc1
Git-commit: 1f61def985d896da5935ae85a69c79fc26c7952f
References: FATE#323952, FATE#322506

QP object is created using rdma_cm api, therefore the destruction
should use the same api for symmetry.

Signed-off-by: Max Gurtovoy <maxg@mellanox.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/rdma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/nvme/host/rdma.c
+++ b/drivers/nvme/host/rdma.c
@@ -493,7 +493,7 @@ static int nvme_rdma_create_queue_ib(str
 	return 0;
 
 out_destroy_qp:
-	ib_destroy_qp(queue->qp);
+	rdma_destroy_qp(queue->cm_id);
 out_destroy_ib_cq:
 	ib_free_cq(queue->ib_cq);
 out_put_dev:
